### ISSUE TITLE
Add the ability to block a rollout

### ISSUE THREAD

**Issue by yertym** [argoproj/argo-rollouts#495]
It would be nice to have the ability to block a rollout from occurring possibly using another custom resource to represent the block, it could store information about why the block was in place.

I can see a hacky way to implement this using an AnalysisRun to check an external source to know whether a block is in place at the start of a rollout but it would be great if the entire setup could be managed by this tool rather than requiring something else to initiate the block.

**Comment by jessesuen** [comment]
I think this can be accomplished by simply having an initial pause step.

**Comment by yertym** [comment]
That would work but I don't think that would convey the intent correctly, a pause is something that can be resumed at will and doesn't contain any metadata about why the pause is occurring. A block is something that requires explanation so that it is obvious when it is acceptable to unblock a particular rollout.

This is only relevant to my implementation but rollouts operates as part of a hands-off pipeline from the moment a commit is merged to master so we don't allow pause steps that don't have a duration specified. A block would work by being separate to the normal steps and just checked prior to the rollout beginning to see if it could commence
